### PR TITLE
Feature/real time flags

### PR DESCRIPTION
**Pull request by gazreese**
Internal review before squashing and sending to the customer

### CODE REVIEW

File: FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt [hunk 1]
```diff
@@ -20,28 +32,74 @@ import com.flagsmith.internal.enqueueWithResult
...
         if (!enableAnalytics) null
         else if (context != null) FlagsmithAnalytics(context, retrofit, analyticsFlushPeriod)
         else throw IllegalArgumentException("Flagsmith requires a context to use the analytics feature")
 
+    private val eventService: FlagsmithEventService? =
+        if (!enableRealtimeUpdates) null
+        else FlagsmithEventService(eventSourceUrl = eventSourceUrl, environmentKey = environmentKey) { event ->
+            if (event.isSuccess) {
+                // First evict the cache otherwise we'll be stuck with the old values
+                cache?.evictAll()
```
Comment (jackforesightmobile) [inline FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt]: Shouldn't we only do this if `lastEventUpdate > lastSeenAt`?

File: FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt [hunk 2]
```diff
@@ -20,28 +32,74 @@ import com.flagsmith.internal.enqueueWithResult
...
+                        }
+                    }
+                }
+            }
+        }
+
+    // The last time we got an event from the SSE stream or via the API
+    private var lastEventUpdate: Double = 0.0
+
+    //TODO: Make this a list of flags?
```
Comment (jackforesightmobile) [inline FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt]: Do you still need to do this?

File: FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt [hunk 3]
```diff
@@ -101,6 +163,20 @@ class Flagsmith constructor(
...
+        eventService.sseEventsFlow.onEach {
+            getFeatureFlags { res ->
+                if (res.isFailure) {
+                    Log.e("Flagsmith", "Error getting flags in SSE stream: ${res.exceptionOrNull()}")
+                    return@getFeatureFlags
+                }
+            }
+        }.catch {
+            Log.e("Flagsmith", "Error in SSE stream: $it")
+        } // launchIn
```
Comment (jackforesightmobile) [inline FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt]: Do we need this comment?

File: FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt (hunk 2 above)
Comment (gazreese) [reply FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt]: Yea I'm just testing this now

File: FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt (hunk 1 above)
Comment (gazreese) [reply FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt]: Done! Everything passes so must be OK

**Pull request merged**